internal/cloudapi/v2/compose.go

```diff
@@ -1441,6 +1445,22 @@ func convertDiskCustomizations(disk *Disk) (*blueprint.DiskCustomization, error)
 	return bpDisk, bpDisk.Validate()
 }
 
+func convertDnfCustomizations(dnf *DNF) *blueprint.DNFCustomization {
+	if dnf == nil {
```

Comment:
```suggestion
	if dnf == nil || dnf.Config == nil {
```

and then drop the first `if` later?